### DISCUSSION TITLE
Feature Request: Percentiles Bucket Aggregation

### DISCUSSION THREAD

**Discussion by tmyl123** [grafana/grafana#58400] [Data Sources]
* What Grafana version are you using?
4.7.0-10031pre1
* What datasource are you using?
Elasticsearch 5.5.1
* What OS are you running grafana on?
Ubuntu 16.04
* What did you do?
Query network devices' counter metric, and use derivative to graph their speed.
* What was the expected result?
All good except for when **we also need 95% percentile on such derivative.**

With Elasticsearch we have [Percentiles Bucket Aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/5.1/search-aggregations-pipeline-percentiles-bucket-aggregation.html#search-aggregations-pipeline-percentiles-bucket-aggregation) , it'll be wonderful if we can directly graph on Grafana, below is our query looks like.

```
"aggs" : {
"tx_per_hour" : {
"date_histogram" : {
"field" : "@timestamp",
"interval" : "hour"
},
"aggs": {
"tx": {
"avg": {
"field": "value"
}
},
"tx_speed": {
"derivative": {
"buckets_path": "tx"
}
}
}
},
"burst_percentile_usage" : {
"percentiles_bucket" : {
"buckets_path" : "tx_per_hour>tx_speed",
"percents" : [95.0]
}
}
}
```

**Comment by tom-howland** [comment]
grafana provides a lovely heat map display for histograms, but the underlying histograms must be summed first. I don't believe this is possible.

![image](https://user-images.githubusercontent.com/26069017/46364052-8ad56600-c629-11e8-93ce-e66342b5d585.png)

**Comment by aocenas** [comment]
Needs to be checked what kind of data elastic returns. I would assume it would be simple time series which should not be a problem to plot.

**Comment by Elfo404** [comment]
This should be pretty straightforward to implement in the current editor, although parsing the results might be a bit tricky.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
